### PR TITLE
WIP: make possible run of tools/unix/generate_styles.sh without python2

### PR DESCRIPTION
**Pull request by Dushistov**
Marked as work in progress, because of it requires merged of https://github.com/organicmaps/kothic/pull/1 ,
then I update link to submodule and only then it would be possible to merge.

### CODE REVIEW

**Review by rtsisyk — APPROVED** [review]

**Comment by biodranik** [comment]
I would also add into all updated/migrated python3 scripts
#!/usr/bin/env python3

File: tools/python/stylesheet/drules_merge.py [hunk 1]
```diff
@@ -106,13 +106,13 @@ def apply_diff(drules, diff):
 
 if __name__ == '__main__':
   if len(sys.argv) <= 3:
-    print 'Usage: {} <drules1.bin> <drules2.bin> <drules_out.bin> [drules_out.txt]'.format(sys.argv[0])
+    print('Usage: {} <drules1.bin> <drules2.bin> <drules_out.bin> [drules_out.txt]'.format(sys.argv[0]))
     sys.exit(1)
 
   drules1 = drules_struct_pb2.ContainerProto()
-  drules1.ParseFromString(open(sys.argv[1]).read())
+  drules1.ParseFromString(open(sys.argv[1], mode='rb').read())
```
Comment (biodranik) [inline tools/python/stylesheet/drules_merge.py:113]: Why binary for ParseFromString?

**Review by biodranik — APPROVED** [review]

File: tools/python/stylesheet/drules_merge.py (hunk 1 above)
Comment (Dushistov) [reply tools/python/stylesheet/drules_merge.py:113]: `open` has different default mode in python 2 vs 3. 
In python3 by default mode to convert newlines and after this new lines conversation (bytes like `0xa`) google protobuf library can not understand bytes received from`open(sys.argv[1]).read()`, while with python2 all work fine.

**Comment by Dushistov** [comment]
> I would also add into all updated/migrated python3 scripts
> #!/usr/bin/env python3

Done

**Review by vng — APPROVED** [review]

**Comment by Dushistov** [comment]
without organicmaps/kothic#1 and switch submodule from omim/kothic to organicmaps/kothic
generate_styles.sh becomes broken, part requires python3 part of code requires python2

File: tools/python/stylesheet/drules_merge.py (hunk 1 above)
Comment (biodranik) [reply tools/python/stylesheet/drules_merge.py:113]: Wow, good to know, thanks!

**Pull request merged**